pkg/homedir/homedir_unix.go

```diff
@@ -94,3 +99,98 @@ func stick(f string) error {
 	m |= os.ModeSticky
 	return os.Chmod(f, m)
 }
+
+var (
+	rootlessConfigHomeDirOnce sync.Once
+	rootlessConfigHomeDir     string
+	rootlessRuntimeDirOnce    sync.Once
+	rootlessRuntimeDir        string
+)
+
+// GetRootlessConfigHomeDir returns the config home directory when running as non root
+func GetRootlessConfigHomeDir() (string, error) {
+	var rootlessConfigHomeDirError error
+
+	rootlessConfigHomeDirOnce.Do(func() {
+		cfgHomeDir := os.Getenv("XDG_CONFIG_HOME")
+		if cfgHomeDir == "" {
+			home := os.Getenv("HOME")
+			resolvedHome, err := filepath.EvalSymlinks(home)
+			if err != nil {
+				rootlessConfigHomeDirError = fmt.Errorf("cannot resolve %s: %w", home, err)
+				return
+			}
+			tmpDir := filepath.Join(resolvedHome, ".config")
+			st, err := os.Stat(tmpDir)
+			if err == nil && int(st.Sys().(*syscall.Stat_t).Uid) == os.Geteuid() && st.Mode().Perm() >= 0o700 {
```

Comment:
The `>=` made my eyes bug out. I understand that `FileMode.Perm` is only the `ugo` portion, but even so this just feels too weird. Is this a common Go idiom? If it isn't, would you mind changing it to `.Perm() & 0o700 == 0o700`?

Reply:
that code is coming from https://github.com/containers/podman/commit/f44b05f6fda6ec91bc21088321da739ce26cfe2d

This is a good occasion to fix it, I'll use your version